Xamarin.Forms.Platform.iOS/Forms.cs

```diff
@@ -334,6 +334,17 @@ public void StartTimer(TimeSpan interval, Func<bool> callback)
 				NSRunLoop.Main.AddTimer(timer, NSRunLoopMode.Common);
 			}
 
+			public void StartTimer(TimeSpan interval, Func<Task<bool>> callback)
+			{
+				NSTimer timer = NSTimer.CreateRepeatingTimer(interval, async t =>
+				{
+					var result = await callback();
+					if (result)
```

Comment:
everywhere you use if (!result), but here if (result). is it correct?

Reply:
This appears to be a typo to me as well. Looking at line 331 from the non `Task` API, it shows that the behavior checks for false and not true. Maybe this should be `if (!result)`

Code snippet from non-async API
https://github.com/xamarin/Xamarin.Forms/blob/75497c66b2de7941c101af2f651c82483d987ca5/Xamarin.Forms.Platform.iOS/Forms.cs#L331-L332

I'm not an iOS expert but is `t.Invalidate();` restarting the timer or re-running the function? If I understand this correctly that function stops the process.

Reply:
Good catch! Yup - I totally forgot the `!` 🙈